src/Account/CommandHandler/CreateIdentityHandler.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License version 3.0
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License version 3.0
+ */
+
+namespace PrestaShop\Module\PsAccounts\Account\CommandHandler;
+
+use Hook;
+use PrestaShop\Module\PsAccounts\Account\Command\CreateIdentityCommand;
+use PrestaShop\Module\PsAccounts\Account\ShopIdentity;
+use PrestaShop\Module\PsAccounts\Api\Client\AccountsClient;
+use PrestaShop\Module\PsAccounts\Provider\OAuth2\Oauth2Client;
+use PrestaShop\Module\PsAccounts\Provider\ShopProvider;
+
+class CreateIdentityHandler
+{
+    /**
+     * @var AccountsClient
+     */
+    private $accountsClient;
+
+    /**
+     * @var Oauth2Client
+     */
+    private $oauth2Client;
+
+    /**
+     * @var ShopProvider
+     */
+    private $shopProvider;
+
+    /**
+     * @var ShopIdentity
+     */
+    private $shopIdentity;
+
+    /**
+     * @param AccountsClient $accountsClient
+     * @param ShopProvider $shopProvider
+     * @param Oauth2Client $oauth2Client
+     * @param ShopIdentity $shopIdentity
+     */
+    public function __construct(
+        AccountsClient $accountsClient,
+        ShopProvider $shopProvider,
+        Oauth2Client $oauth2Client,
+        ShopIdentity $shopIdentity
+    ) {
+        $this->accountsClient = $accountsClient;
+        $this->shopProvider = $shopProvider;
+        $this->oauth2Client = $oauth2Client;
+        $this->shopIdentity = $shopIdentity;
+    }
+
+    /**
+     * @param CreateIdentityCommand $command
+     *
+     * @return void
+     */
+    public function handle(CreateIdentityCommand $command)
+    {
+        // FIXME: remove that test
+        // FIXME: migration from v7 -> v8 event modeling
+        // - cleanup configuration storage
+        // - identify shop (when ?) -> be sure we send version with it & when to trigger it ?
+        // - UX associated ?
+        // - Migrate routes using user token
+        if (!$this->oauth2Client->exists()) {
```

Comment:
I think it's better to invert this condition and apply the return early pattern. For me, it's more readable to read a code without much imbrication or _else_ clause